hs_core/management/utils.py

```diff
@@ -654,54 +657,44 @@ def test(self):
-        msg = "Resource with id {} not found in Django Resources".format(resource.short_id)
-        if log_errors:
-            logger.error(msg)
-        if echo_errors:
-            print(msg)
-        if return_errors:
-            errors.append(msg)
-            ecount = ecount + 1
-        return errors, ecount
+def repair_resource(resource, logger):
 
-    print("REPAIRING RESOURCE {}".format(resource.short_id))
+    print("CHECKING IF RESOURCE {} NEEDS REPAIR".format(resource.short_id))
+    now = timezone.now()
 
     # ingest any dangling iRODS files that you can
     # Do this before check because otherwise, errors get printed twice
-    # TODO: This does not currently work properly for composite resources
-    # if resource.resource_type == 'CompositeResource' or \
     if resource.resource_type == 'CompositeResource':
-        _, count = ingest_irods_files(resource,
-                                      logger,
-                                      stop_on_error=False,
-                                      echo_errors=True,
-                                      log_errors=False,
-                                      return_errors=False)
-        if count:
+        _, ingest_count = ingest_irods_files(resource,
+                                             logger,
+                                             stop_on_error=False,
+                                             echo_errors=True,
+                                             log_errors=False,
+                                             return_errors=False)
+        if ingest_count:
             print("... affected resource {} has type {}, title '{}'"
                   .format(resource.short_id, resource.resource_type,
                           resource.title))
 
-    _, count = check_irods_files(resource,
-                                 stop_on_error=False,
-                                 echo_errors=True,
-                                 log_errors=False,
-                                 return_errors=False,
-                                 clean_irods=False,
-                                 clean_django=True,
-                                 sync_ispublic=True)
-    if count:
+            resource.repaired = now
+
+    _, check_count = check_irods_files(resource,
+                                       stop_on_error=False,
+                                       echo_errors=True,
+                                       log_errors=False,
+                                       return_errors=False,
+                                       clean_irods=False,
+                                       clean_django=True,
+                                       sync_ispublic=True)
+    if check_count:
         print("... affected resource {} has type {}, title '{}'"
               .format(resource.short_id, resource.resource_type,
                       resource.title))
+        resource.repaired = now
+
+    resource.files_checked = now
+    resource.save()
```

Comment:
If we are repairing many resource this may be better `resource.save(update_fields=["repaired", "files_checked"])`

Reply:
Good call, will fix this

Reply:
fixed